sklearn/ensemble/_bagging.py

```diff
@@ -135,8 +135,13 @@ def _parallel_build_estimators(
                 not_indices_mask = ~indices_to_mask(indices, n_samples)
                 curr_sample_weight[not_indices_mask] = 0
 
-            estimator_fit(X[:, features], y, sample_weight=curr_sample_weight)
+            # Indicator of if indexing is necessary
+            require_indexing = bootstrap_features or max_features != n_features
```

Comment:
I think we can define this outside the loop. This optimization is also useful for the `else:` case a few lines below. Something like:

```python
# outside of loop
requires_feature_indexing = bootstrap_features or max_features != n_features

# in loop
    ...
    X_ = X[:, features] if requires_feature_indexing else X
    estimator_fit(X_, y, sample_weight=curr_sample_weight)
else:
    X_ = X[indices][:, features] if requires_feature_indexing else X[indices]
    estimator_fit(X_, y[indices])
```

Reply:
Updated as suggested :)